### PR TITLE
[PRT-280] Complete mint functionality 

### PR DESCRIPTION
**Pull request by kargakis**
Depends on https://github.com/regen-network/toucan-bridge/pull/5

### CODE REVIEW

File: contracts/ToucanRegenBridge.sol [hunk 1]
```diff
@@ -31,6 +35,23 @@ contract ToucanRegenBridge is Ownable, Pausable {
...
+    //      Modifiers
+    // ----------------------------------------
+
+    modifier isRegenAddress(bytes calldata account) {
+        // verification: checking if account starts with "regen1"
+        require(account.length >= 44, "regen address is at least 44 characters long");
+        bytes memory prefix = "regen1";
+        for (uint8 i = 0; i < 6; ++i)
+            require(prefix[i] == account[i], "regen address must start with 'regen1'");
+        _;
```
Comment (pheuberger) [inline contracts/ToucanRegenBridge.sol:48]: just curious, what does this line do?

File: contracts/ToucanRegenBridge.sol [hunk 2]
```diff
@@ -79,27 +101,20 @@ contract ToucanRegenBridge is Ownable, Pausable {
...
+        tco2Limits[tco2] -= amount;
 
-    function isRegenAddress(bytes memory recipient) internal pure returns (bool) {
-        // verification: checking if recipient starts with "regen1"
-        require(recipient.length >= 44, "regen address is at least 44 characters long");
-        bytes memory prefix = "regen1";
-        for (uint8 i = 0; i < 6; ++i)
-            require(prefix[i] == recipient[i], "regen address must start with 'regen1'");
-        return true;
+        emit Issue(sender, recipient, address(tco2), amount);
```
Comment (pheuberger) [inline contracts/ToucanRegenBridge.sol:117]: is wrapping in address necessary here? `tco2` is already an address.

File: test/ToucanRegenBridge.js [hunk 3]
```diff
@@ -73,9 +73,11 @@ describe("Token contract", function () {
 	describe("Issue TCO2 tokens", function () {
```
Comment (pheuberger) [inline test/ToucanRegenBridge.js:73]: should we also add a test that tests the happy path?

Comment (kargakis) [reply test/ToucanRegenBridge.js:73]: Need to merge https://github.com/ToucanProtocol/tokenizer/pull/1402 first and update the submodule in this repo before we can test the happy path.

File: contracts/ToucanRegenBridge.sol (hunk 2 above)
Comment (kargakis) [reply contracts/ToucanRegenBridge.sol:117]: Not needed, will remove

File: contracts/ToucanRegenBridge.sol (hunk 1 above)
Comment (kargakis) [reply contracts/ToucanRegenBridge.sol:48]: The rest of the function code that is modified is executed.

Comment (kargakis) [reply contracts/ToucanRegenBridge.sol:48]: https://www.tutorialspoint.com/solidity/solidity_function_modifiers.htm

File: test/ToucanRegenBridge.js (hunk 3 above)
Comment (pheuberger) [reply test/ToucanRegenBridge.js:73]: gotcha!

**Review by pheuberger — APPROVED** [review]

**Pull request merged**